ddtrace/monkey.py

```diff
@@ -219,6 +224,10 @@ def patch_module(module, raise_errors=True):
         return False
 
 
+@deprecated(
+    message="This function will be removed.",
+    version="1.0.0",
+)
 def get_patched_modules():
```

Comment:
not sure why I missed it before.

these functions are still used internally, so we don't want to give this deprecation warning when we use them, only when someone else does.

To handle this we will need to do something like the following:

```python
# Rename the existing patch_module -> _patch_module
def _patch_module(module, raise_errors=True):
    ...


@deprecated("message", version="1.0.0")
def patch_module(module, raise_errors=True):
    return _patch_module(module, raise_errors=raise_errors)
```

Then we would need to update all internal calls to `patch_module` -> `_patch_module`

Reply:
There is currently a _path_module function beneath the deprecated functions:

```
def _patch_module(module):
    # type: (str) -> bool
    """_patch_module will attempt to monkey patch the module.

    Returns if the module got patched.
    Can also raise errors if it fails.
    """
    path = "ddtrace.contrib.%s" % module
    with _LOCK:
        if module in _PATCHED_MODULES and module not in _PATCH_ON_IMPORT:
            log.debug("already patched: %s", path)
            return False

```

Reply:
@Kyle-Verhoog or @Yun-Kim any thoughts?

Reply:
It looks like `patch_module()` currently is just a wrapper for `_patch_module()` that includes error handling. Just my two cents on ideas what we can do with it while handling it the way Brett suggested:

- One option could be to move `_patch_module()` to be a nested function within `patch_module()` and rename `patch_module()` -> `_patch_module()` since that's the only place we ever use it. 

- Another option that I see might be simpler is to rename `_patch_module()` --> `_attempt_patch_module()` and `patch_module()` -> `_patch_module()`.

Reply:
> Another option that I see might be simpler is to rename _patch_module() --> _attempt_patch_module() and patch_module() -> _patch_module().

This seems like a decent option, especially since encapsulating the logic in one place is nice, since `_patch_module()` is called from 2 (?) places.

Reply:
I can only see `_patch_module()` being called by `patch_module()`, where else is it called from?

Reply:
oh, sorry, I meant `patch_module` is called from multiple places.